lib/matplotlib/testing/compare.py

```diff
@@ -141,38 +140,82 @@ def _shlex_quote_bytes(b):
             else b"'" + b.replace(b"'", b"'\"'\"'") + b"'")
 
 
-class _SVGConverter(object):
+class _ConverterError(Exception):
+    pass
+
+
+class _Converter(object):
     def __init__(self):
         self._proc = None
-        # We cannot rely on the GC to trigger `__del__` at exit because
-        # other modules (e.g. `subprocess`) may already have their globals
-        # set to `None`, which make `proc.communicate` or `proc.terminate`
-        # fail.  By relying on `atexit` we ensure the destructor runs before
-        # `None`-setting occurs.
+        # Explicitly register deletion from an atexit handler because if we
+        # wait until the object is GC'd (which occurs later), then some module
+        # globals (e.g. signal.SIGKILL) has already been set to None, and
+        # kill() doesn't work anymore...
         atexit.register(self.__del__)
 
-    def _read_to_prompt(self):
-        """Did Inkscape reach the prompt without crashing?
-        """
-        stream = iter(functools.partial(self._proc.stdout.read, 1), b"")
-        prompt = (b"\n", b">")
-        n = len(prompt)
-        its = itertools.tee(stream, n)
-        for i, it in enumerate(its):
-            next(itertools.islice(it, i, i), None)  # Advance `it` by `i`.
+    def __del__(self):
+        if self._proc:
+            self._proc.kill()
+            self._proc.wait()
+            for stream in filter(None, [self._proc.stdin,
+                                        self._proc.stdout,
+                                        self._proc.stderr]):
+                stream.close()
+            self._proc = None
+
+    def _read_until(self, terminator):
+        """Read until the prompt is reached."""
+        terminator = [six.int2byte(c) for c in six.iterbytes(terminator)]
```

Comment:
If you would like to eliminate the `six` import while you are in the neighborhood, you could replace this line with
```
terminator = [terminator[i:i+1] for i in range(len(terminator))]
```
or omit that line, so that terminator remains a bytes object, and fix a couple lines lower down to do everything with bytes objects instead of a list of single bytes objects.

Reply:
done